### PR TITLE
Made small edits to make transferring a compiled library to the simulator easier

### PR DESCRIPTION
**Pull request by jessberg**
These are really small edits, just library names and such, to make the compiler work with the simulator.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -160,7 +155,6 @@ fn main() {
         );
     } else {
         let filter_handlebars_cc = bin_dir.join("filter.cc.handlebars");
-        output_name.set_extension("cc");
```
Comment (fruffy) [inline src/main.rs:163]: Something went wrong here. This is incorrect.

File: src/main.rs [hunk 2]
```diff
@@ -128,18 +128,13 @@ fn main() {
     // Use the information in the code generator code_gen and format it, using handlebars
     // to a template with all the basic filter information enclosed
     let c_mode = matches.value_of("compilation_mode").unwrap();
-    let mut output_name = PathBuf::from(matches.value_of("output").unwrap());
+    let output_name = PathBuf::from(matches.value_of("output").unwrap());
```
Comment (fruffy) [inline src/main.rs:131]: This should be mutable.

**Review by fruffy — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 1 above)
Comment (jessberg) [reply src/main.rs:163]: Fixed

File: src/main.rs (hunk 2 above)
Comment (jessberg) [reply src/main.rs]: Fixed

**Review by fruffy — APPROVED** [review]

**Pull request merged**